internal/dyngo/operation.go

```diff
@@ -0,0 +1,126 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package dyngo
+
+type (
+	InstrumentationDescriptor struct {
+		Title           string
+		Instrumentation Instrumentation
+	}
+
+	Instrumentation interface{ isInstrumentation() }
+
+	OperationInstrumentation struct {
+		EventListener EventListener
+	}
+
+	FunctionInstrumentation struct {
+		Symbol   string
+		Prologue interface{}
+	}
+)
+
+func (OperationInstrumentation) isInstrumentation() {}
+func (FunctionInstrumentation) isInstrumentation()  {}
+
+func Register(descriptors ...InstrumentationDescriptor) (ids []EventListenerID) {
+	for _, desc := range descriptors {
+		switch actual := desc.Instrumentation.(type) {
+		case OperationInstrumentation:
+			ids = append(ids, root.Register(actual.EventListener)...)
+		}
+	}
+	return ids
+}
+
+func Unregister(ids []EventListenerID) {
+	root.Unregister(ids)
+}
+
+type (
+	Option interface {
+		apply(s *Operation)
+	}
+
+	optionFunc func(s *Operation)
+)
+
+func (f optionFunc) apply(s *Operation) {
+	f(s)
+}
+
+func WithParent(parent *Operation) Option {
+	return optionFunc(func(op *Operation) {
+		op.parent = parent
+	})
+}
+
+var root = newOperation()
+
+type Operation struct {
+	parent *Operation
+	eventManager
```

Comment:
It looks to me like `eventManager` should be merged into `Operation`. I don't see a reason to have a separate `eventManager`.

Their method definitions are already spread between the files (some `Operation` methods defined in events.go and an `eventManager` method defined here) which further indicates this.

Reply:
Done in 8a6dba0